backend/src/schema/resolvers/searches.spec.js

```diff
@@ -0,0 +1,452 @@
+            describe('query contains the title of the first post', () => {
+              it('finds both posts', async () => {
+                variables = { query: 'beitrag' }
+                await expect(query({ query: searchQuery, variables })).resolves.toMatchObject({
+                  data: {
+                    findResources: expect.arrayContaining([
+                      {
+                        __typename: 'Post',
+                        id: 'a-post',
+                        title: 'Beitrag',
+                        content: 'Ein erster Beitrag',
+                      },
+                      {
+                        __typename: 'Post',
+                        id: 'b-post',
+                        title: 'Aufruf',
+                        content: 'Jeder sollte seinen Beitrag leisten.',
+                      },
+                    ]),
+                  },
+                })
+              })
+            })
+          })
+
+          describe('a post that contains a hyphen between two words and German quotation marks', () => {
+            describe('hyphens in query', () => {
+              it('will be treated as ordinary characters', async () => {
+                variables = { query: 'tee-ei' }
+                await expect(query({ query: searchQuery, variables })).resolves.toMatchObject({
+                  data: {
+                    findResources: [
+                      {
+                        __typename: 'Post',
+                        id: 'g-post',
+                        title: 'Zusammengesetzte Wörter',
+                        content: `Ein Bindestrich kann zwischen zwei Substantiven auch dann gesetzt werden, wenn drei gleichlautende Buchstaben aufeinandertreffen. Das ist etwa bei einem „Teeei“ der Fall, das so korrekt geschrieben ist. Möglich ist hier auch die Schreibweise mit Bindestrich: Tee-Ei.`,
+                      },
+                    ],
+                  },
+                })
+              })
+            })
+
+            describe('German quotation marks in query', () => {
+              it('will be treated as ordinary characters', async () => {
+                variables = { query: '„teeei“' }
+                await expect(query({ query: searchQuery, variables })).resolves.toMatchObject({
+                  data: {
+                    findResources: [
+                      {
+                        __typename: 'Post',
+                        id: 'g-post',
+                        title: 'Zusammengesetzte Wörter',
+                        content: `Ein Bindestrich kann zwischen zwei Substantiven auch dann gesetzt werden, wenn drei gleichlautende Buchstaben aufeinandertreffen. Das ist etwa bei einem „Teeei“ der Fall, das so korrekt geschrieben ist. Möglich ist hier auch die Schreibweise mit Bindestrich: Tee-Ei.`,
+                      },
+                    ],
+                  },
+                })
+              })
+            })
+          })
```

Comment:
I'm confused by these two test cases... is there some docs you can link to that explain why this was an issue in the past?
I went to the link you provided https://lucene.apache.org/core/8_3_1/queryparser/org/apache/lucene/queryparser/classic/package-summary.html#Escaping_Special_Characters , but I'm not sure where the German quotation marks fit into the equation.

Also looking at the tests, it's not clear. The first test searches for `tee-ei`, which the content does contain, so I would expect it to appear in the list, but maybe it wasn't because of the German quotations, somehow?

Reply:
The quotation marks are a testcase for unicode chars. I will put that in the description.